### ISSUE TITLE
SRVConnector keeps reconnecting on non-SRV responses

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1908]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#1908|
|Type|enhancement|
|Created|2006-07-10 15:09:17Z|

When a nameserver sends out CNAME records in response to a SRV query (this happens when you have *.domain as a CNAME to a real host), the reference to `answers[0].payload.target` raises an exception in the `_cbGotServers` method of `t.n.srvconnect.SRVConnector`. The callback always expects SRV records in the response, and doesn't check for it. When the exception is raised, the query is repeated.

<details><summary>Searchable metadata</summary>

```
trac-id__1908 1908
type__enhancement enhancement
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1152544157000000 1152544157000000
changetime__1187945105000000 1187945105000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

(In [21155]) Add test cases for the NXDOMAIN reply on SRV lookup.

This also addresses [#1908](https://github.com/twisted/twisted/issues/1908).

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm set status to `closed`|
|-|-|

(In [21159]) Make SRVConnector work with bad results and NXDOMAIN responses.

Author: ralphm
Reviewer: therve
Fixes [#1908](https://github.com/twisted/twisted/issues/1908), [#2777](https://github.com/twisted/twisted/issues/2777).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
